src/io/writer_internal.cpp

```diff
@@ -73,74 +80,71 @@ void WriterInternal::WriteHeader(las::LasHeader &head14)
     {
         laz_header.vlr_count++;
     }
+    if (!file_->GetWkt().empty())
+    {
+        laz_header.vlr_count++;
+    }
+
+    // This is always zero since we are limited to point format 6-8
+    laz_header.point_count = 0;
```

Comment:
you still need to set the legacy point_count according to the logic I wrote per the las spec